src/main/java/com/xpanxion/java/springboot/da1/demo/controller/student1/BookController1.java

```diff
@@ -27,18 +33,25 @@ public Book getBook(@PathVariable("id") Integer id) {
 
     @PostMapping("student1/api/v1/bookstores/{bookstoreId}/books")
     public void insertBook(@PathVariable("bookstoreId") Integer bookstoreId, @RequestBody Book newBook) {
-        dataAccess1.insertBook(newBook);
-        dataAccess1.addToBookstore(newBook, bookstoreId);
+        int bookId = dataAccess1.insertBook(newBook);
+        dataAccess1.addToBookstore(newBook, bookstoreId, bookId);
     }
 
     @Service
     public static class DataAccess {
         @Autowired
         private JdbcTemplate jdbcTemplate;
+
+        @Autowired
+        private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+        private KeyHolder holder = new GeneratedKeyHolder();
```

Comment:
Let's move this into the function.  I don't think there's a good reason to make this a private data member.